ipn/ipnlocal/local_test.go

```diff
@@ -1885,6 +1901,47 @@ func TestSetExitNodeIDPolicy(t *testing.T) {
 	}
 }
 
+func TestUpdateNetmapDeltaAutoExitNode(t *testing.T) {
+	b := newTestLocalBackend(t)
+	policy := "auto"
+	msh := &mockSyspolicyHandler{
+		t: t,
+		stringPolicies: map[syspolicy.Key]*string{
+			syspolicy.ExitNodeID: &policy,
+		},
+	}
+	syspolicy.SetHandlerForTest(t, msh)
+	b.netMap = &netmap.NetworkMap{}
+	for i := range 2 {
+		b.netMap.Peers = append(b.netMap.Peers, (&tailcfg.Node{ID: (tailcfg.NodeID(i) + 1), StableID: tailcfg.StableNodeID(fmt.Sprint(i + 1)), Cap: 26}).View())
+	}
+	b.updatePeersFromNetmapLocked(b.netMap)
+	b.lastSuggestedExitNode = tailcfg.StableNodeID("1")
+	b.SetPrefsForTest(b.pm.CurrentPrefs().AsStruct())
+	someTime := time.Unix(123, 0)
+	muts, ok := netmap.MutationsFromMapResponse(&tailcfg.MapResponse{
+		PeersChangedPatch: []*tailcfg.PeerChange{
+			{
+				NodeID: 1,
+				Online: ptr.To(false),
+			},
+			{
+				NodeID: 2,
+				Online: ptr.To(true),
+			},
+		},
+	}, someTime)
+	if !ok {
+		t.Fatal("netmap.MutationsFromMapResponse failed")
+	}
+
+	b.UpdateNetmapDelta(muts)
```

Comment:
Check that the return value is what you expect

There should also probably be a case where stuff changes but not the current auto exit node, and validate that still handles it fine.